zerver/lib/users.py

```diff
@@ -284,6 +285,27 @@ def access_user_by_id(
     return access_user_common(target, user_profile, allow_deactivated, allow_bots, for_admin)
 
 
+def access_user_by_id_including_cross_realm(
+    user_profile: UserProfile,
+    target_user_id: int,
+    *,
+    allow_deactivated: bool = False,
+    allow_bots: bool = False,
+    for_admin: bool,
+) -> UserProfile:
+    """Master function for accessing another user by ID in API code;
+    verifies the user ID (including cross realm users) is in the same
+    realm, and if requested checks for administrative privileges, with
+    flags for various special cases.
+    """
```

Comment:
Never copy-paste docstrings from a function that does something different! I'll rewrite this one -- but one should be very careful with doing copy-paste in a production code base, as it's very easy to end up submitting something wrong that you'd never have generated were you not using copy-pasting.

As a more meta-level comment, I think it's worth taking more time on each of your PRs to review your own work carefully. It can sometimes be OK to be sloppy with things like this when prototyping, but if you do that, you need to figure out a set of habits that avoid asking reviewers to spend their limited time reviewing the not-yet-cleaned-up version of that. The most important metric for a software developer in an open source project is how able they are to create PRs that are easy to review and usually correct the first time, and I think you're at the point where you should start focusing on that.